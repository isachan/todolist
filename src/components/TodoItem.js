import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
const TodoItem = ({ item, onCheck, onDelete }) => {
  return (
    <div className='todoItem' onClick={onCheck}>
      <div className='todoItem__item'>
        {item.checked ? (
          <p
            data-testid='item-content'
            style={{ textDecoration: 'line-through' }}
          >
            {item.item}
          </p>
        ) : (
          <p data-testid='item-content'>{item.item}</p>
        )}
        <button
          className='todoItem__delete-icon'
          data-testid='delete-button'
          onClick={onDelete}
        >
          X
        </button>
      </div>
      <div className='todoItem__date'>
        <p>{moment(item.dateCreated).format('ddd, MMM do, h:mmA')}</p>
      </div>
    </div>
  );
};

TodoItem.propTypes = {
  item: PropTypes.object.isRequired,
  onCheck: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired
};

export default TodoItem;
