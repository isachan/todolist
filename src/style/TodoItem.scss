.todoItem {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem 0;

  > * {
    width: 100%;
  }
  &__delete-icon {
    height: 24px;
    width: 24px;
  }

  &__item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }
  &__date {
    display: flex;
    flex-direction: row;
    font-size: 12px;
  }
}
