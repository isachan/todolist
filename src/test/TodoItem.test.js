import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoItem from '../components/TodoItem';

import { render, screen, fireEvent } from '@testing-library/react';

const dummyItem = {
  item: 'wash dish',
  checked: false,
  dateCreated: new Date()
};

describe('TodoItem Component', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  it('renders correctly', () => {
    const component = shallow(
      <TodoItem item={dummyItem} onCheck={() => {}} onDelete={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });

  it('clicking on X delete button removes record', () => {
    const { getByTestId } = render(
      <TodoItem item={dummyItem} onCheck={() => {}} onDelete={() => {}} />
    );
    const deleteButton = getByTestId('delete-button');
    // console.log('FIRST!!!', deleteButton);
    fireEvent.click(deleteButton);
    // need to write function to test data from firebase online
    screen.getByText('X');

    expect(screen.findByText('X').toBeVisible);
    // console.log('SECONDDD!!!', deleteButton);
  });
});
