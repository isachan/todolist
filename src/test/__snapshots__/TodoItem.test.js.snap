// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component renders correctly 1`] = `ShallowWrapper {}`;
